db/migrations/20220608090500_remove_duplicated_tenant_ids_org_ids.go

```diff
@@ -0,0 +1,251 @@
+package migrations
+
+import (
+	"fmt"
+
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	"github.com/RedHatInsights/sources-api-go/model"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+// duplicateTenantMark is the mark that will be left on the "external_tenant" and "org_id" columns for those tenants
+// that are duplicated. The idea is to prepare the tenants to be able to apply the unique constraint on the columns
+// right after.
+const duplicateTenantMark = "processed-duplicate-tenant-of-%s-%d"
+
+// RemoveDuplicatedTenantIdsOrgIds fetches all the duplicated tenants, makes sure that they are indeed duplicated, and
+// unifies all the subresources in a single tenant before deleting the duplicated one.
+func RemoveDuplicatedTenantIdsOrgIds() *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "20220608090500",
+		Migrate: func(db *gorm.DB) error {
+			logging.Log.Info(`Migration "remove duplicated tenants" started`)
+			defer logging.Log.Info(`Migration "remove duplicated tenants" ended`)
+
+			// Updatable tables contains all the table names which have the "tenant_id" column which must be updated.
+			updatableTables := []string{
+				"application_authentications",
+				"applications",
+				"authentications",
+				"endpoints",
+				"sources",
+				"source_rhc_connections",
+				"users",
+			}
+
+			// This query selects all the tenants which have duplicated "external_tenant" or "org_id" column values.
+			duplicatesSql := `
+				SELECT
+					"t"."id", "t"."name", "t"."description", "t"."external_tenant", "t"."org_id", "t"."created_at", "t"."updated_at"
```

Comment:
do we need all columns here ?

Reply:
You are right, we can probably narrow it down just to the "id", "external_tenant" and "org_id" columns

Reply:
Well now that I think about it, I log the entire tenant object with a `"%v#` on the logging statement. So maybe we can leave it like this, or simply a `*`?